app/lib/import_export.rb

```diff
@@ -345,10 +672,10 @@ def latest_transfers(patients)
         ]
   end
 
-  # 2 Latest laboratories of each patient
-  def latest_laboratories(patients)
+  # Gets a hash of the 2 latest laboratories of each patient
+  def get_patients_laboratories(patients)
     latest_labs = Hash[patients.pluck(:id).map { |id| [id, {}] }]
-    Laboratory.where(patient_id: patients.pluck(:id)).order(report: :desc).each do |lab|
+    Laboratory.where(patient_id: patients.pluck(:id)).order(specimen_collection: :desc).each do |lab|
       if !latest_labs[lab.patient_id].key?(:first)
         latest_labs[lab.patient_id][:first] = {
           lab_type: lab[:lab_type],
```

Comment:
I frustratingly cannot comment on the below code because it's unchanged, but can we add a return at the end of the elseif where we find the second lab? That way we don't need to loop through the rest of them once we find the second one.

Reply:
I think that might not be possible because we're looping through all the labs of all the patients at the same time, not just the labs of a single patient at a time, so returning after finding the second lab for one patient would essentially skip/end the process of finding the rest of the labs for all the other patients

Reply:
Unfortunately I don't think it would be easy to do this all in SQL, particularly because I'd rather not mess with the `sql_mode=only_full_group_by` global; however I think you can tweak line 679 to make this cleaner, using something like `.group_by(&:patient_id).transform_values { |v| v.take(2) }`, which would return a hash of patient ids, with values of array containing up to two lab results per patient.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/562/commits/18f2cd1c1295b23fca3eb469bf542092217c6acb#diff-c34521fcd0fdc2034d456886aae8fd35299fe908ab88d8e49c60d0b4d39ab26bR676